File: training-data/php/perfect_number.php
<?php

function divisorSum(int $n): int
{
    if ($n < 2) {
        return 0;
    }
    $sum = 1;
    for ($i = 2; $i * $i <= $n; $i++) {
        if ($n % $i === 0) {
            $sum += $i;
            if ($i !== intdiv($n, $i)) {
                $sum += intdiv($n, $i);
            }
        }
    }
    return $sum;
}

function classify(int $n): string
{
    return match (divisorSum($n) <=> $n) {
        0 => 'perfect',
        1 => 'abundant',
        -1 => 'deficient',
    };
}

echo 'perfect < 10000: ', implode(', ', array_filter(range(2, 9999), fn ($n) => divisorSum($n) === $n)), PHP_EOL;
foreach ([12, 28, 35] as $n) {
    echo "$n is ", classify($n), PHP_EOL;
}

File: training-data/python/perfect_number.py
def divisor_sum(n):
    if n < 2:
        return 0
    total = 1
    i = 2
    while i * i <= n:
        if n % i == 0:
            total += i
            if i != n // i:
                total += n // i
        i += 1
    return total


def classify(n):
    s = divisor_sum(n)
    return "perfect" if s == n else "abundant" if s > n else "deficient"


if __name__ == "__main__":
    print([n for n in range(2, 10000) if divisor_sum(n) == n])
    for n in (12, 28, 35):
        print(n, classify(n))
